### DISCUSSION TITLE
I would like to use env variable in the .releaserc.yaml file

### DISCUSSION THREAD

**Discussion by hpe-ykoehler** [semantic-release/semantic-release#2837] [Configuration]
### New feature motivation

My jenkins will build a new docker image for each build, i use the label build-${env.BUILD_NUMBER} in jenkins, I would then like semantic-release to publish this image, so I tell it to:

```
{
  "branches": [
    "mainline",
    { name: "devel", prerelease: 'alpha'}
  ],
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    ["@semantic-release/exec", {
      "prepareCmd": "echo '${nextRelease.version}' > version",
      "publishCmd": "./build.sh -u"
    }],
    ["@semantic-release/git", {
      "assets": ["CHANGELOG.md", "version"],
      "message": "chore(release): ${nextRelease.version}"
    }],
    ["semantic-release-docker", {
      "name": "dev-build-image:build-${BUILD_NUMBER}",
      "registryUrl": "https://docker.arubacorp.net:8443"
    }]
  ]
}
```

But the semantic-release doesn't seem to accept thte ${BUILD_NUMBER} is an env variable and just pass that as a string.

```
22:18:32  [2:18:32 AM] [semantic-release] › ℹ  Start step "publish" of plugin "semantic-release-docker"
22:18:32  [2:18:32 AM] [semantic-release] [semantic-release-docker] › ℹ  Pushing version dev-build-image:build-${BUILD_NUMBER}:1.0.1 to docker hub
22:18:32  [2:18:32 AM] [semantic-release] › ✔  Completed step "publish" of plugin "semantic-release-docker"
22:18:32  [2:18:32 AM] [semantic-release] › ℹ  Start step "success" of plugin "@semantic-release/exec"
22:18:32  [2:18:32 AM] [semantic-release] › ✔  Completed step "success" of plugin "@semantic-release/exec"
22:18:32  [2:18:32 AM] [semantic-release] › ✔  Published release 1.0.1 on default channel
```

Is there a way for me to use an external env variable within the .releaserc.yaml script?

### New feature description

accept env variable from the calling environment?

**Comment by travi** [comment]
your best bet for leveraging an environment variable would be to use a javascript file rather than yaml for the config. that way you can reference using `process.env.VARIABLE_NAME`

**Comment by hpe-ykoehler** [comment]
Do you imply I could use ${process.env.BUILD_NUMBER} ?

but since release-semantic support ${nextrelease} and GIT env, why are the other env variable filtered out?

**Reply by travi** [reply]
values like `${nextRelease.version}` are template variables that are populated by semantic-release for plugins for each lifecycle step, not environment variables. more details about what is populated for each lifecycle for plugins can be found [in our docs](https://semantic-release.gitbook.io/semantic-release/developer-guide/plugin#context-object-keys-by-lifecycle). 

gaining access to the environment variables at runtime is independent from this and even from semantic-release altogether. that is why the config needs to be formatted as js
